cmd/dagger/flags.go

```diff
@@ -449,8 +450,14 @@ func (v *secretValue) Get(ctx context.Context, c *dagger.Client, _ *dagger.Modul
 		plaintext = string(filePlaintext)
 
 	case commandSecretSource:
-		// #nosec G204
-		stdoutBytes, err := exec.CommandContext(ctx, "sh", "-c", v.sourceVal).Output()
+		var stdoutBytes []byte
+		var err error
+		if runtime.GOOS == "windows" {
+			stdoutBytes, err = exec.CommandContext(ctx, "pwsh", "-c", v.sourceVal).Output()
```

Comment:
If we are using `pwsh` we may need to stay Dagger on windows `requires` powershell core (the xplat version of powershell) to be installed. And im not sure if Powershell Core already comes with windows 10/11 by default. Though im sure in the future it will replace `Windows PowerShell`... and thus that is my thought about instead of powershell, maybe cmd.exe is safer, because `cmd.exe` is the native windows command prompt.

Reply:
What do you think about try lookup `pwsh` first, then `powershell.exe` if `pwsh` is not present?

Reply:
> What do you think about try lookup `pwsh` first, then `powershell.exe` if `pwsh` is not present?

If a user wants to use powershell, for some specific reason....

`--my-arg=cmd:"pwsh.exe -c '$(custom-cli-tool)'"`

This means cmd.exe will execute pwsh for the user. Maybe it should default to `cmd.exe`? 

Would be good to get more feedback.

Reply:
Yeah, i am now thinking `cmd.exe` is best option. If a user need to change this, they can execute or start `powershell.exe` or `pwsh.exe` from the default `cmd.exe` which is used for `cmd:`. This way, dagger stays neutral and uses the native cmd prompt. It also mean we don't need to 'check' or 'fallback'. So the implementation stays simple.

Reply:
![image](https://github.com/user-attachments/assets/82a33e8d-5a74-41ed-b90f-535633a009b3)

Reply:
![image](https://github.com/user-attachments/assets/e5bd1a0b-cf08-46cc-9b1a-16af9be80277)

Reply:
![image](https://github.com/user-attachments/assets/92667182-3338-4f8b-93cb-cafb95bdc15b)

Reply:
> So the implementation stays simple.

Agreed. Let me try.

Reply:
Using cmd.exe also works 🎉 

```
PS C:\Users\WingyMomo\src\github.com\dagger\dagger\potato> cat .\main.go
package main

import (
        "context"

        "github.com/dagger/dagger/potato/internal/dagger"
)

type Potato struct{}

func (m *Potato) Insecure(ctx context.Context, token *dagger.Secret) (string, error) {
        return token.Plaintext(ctx)
}
PS C:\Users\WingyMomo\src\github.com\dagger\dagger\potato> ..\bin\dagger call insecure --token='cmd:echo shhh'
Setup tracing at https://dagger.cloud/traces/setup. To hide: export NOTHANKS=1

✔ connect 0.2s
✔ initialize 1.3s
✔ prepare 0.0s
✔ setSecret(name: "a9e4d508131009f921974e329a194d65b473daa4b029a9d9c6c444a0aa663895"): Secret! 0.0s
✔ potato: Potato! 0.0s
✔ Potato.insecure(
    token: ✔ setSecret(name: "a9e4d508131009f921974e329a194d65b473daa4b029a9d9c6c444a0aa663895"): Secret! 0.0s
  ): String! 0.3s

shhh
PS C:\Users\WingyMomo\src\github.com\dagger\dagger\potato> ..\bin\dagger call insecure --token='cmd:pwsh.exe -Command echo shhh'
Setup tracing at https://dagger.cloud/traces/setup. To hide: export STOPIT=1

✔ connect 0.2s
✔ initialize 1.2s
✔ prepare 0.3s
✔ setSecret(name: "a9e4d508131009f921974e329a194d65b473daa4b029a9d9c6c444a0aa663895"): Secret! 0.0s
✔ potato: Potato! 0.0s
✔ Potato.insecure(
    token: ✔ setSecret(name: "a9e4d508131009f921974e329a194d65b473daa4b029a9d9c6c444a0aa663895"): Secret! 0.0s
  ): String! 0.4s                                                                                                                                                                                                 
shhh
```

Reply:
Nice, i think this is safest way for now.